mail/opensmtpd/Portfile

```diff
@@ -0,0 +1,104 @@
+
+checksums           rmd160  740a0904594f47caa1261b4be3bc7de8165678e7 \
+                    sha256  51bf05312fe56764748faab9e63170bfbb7c7d84a37c4117b62b2681eb3a4d17 \
+                    size    860189
+
+depends_build       port:bison
+
+depends_lib         port:libevent \
+                    path:lib/libssl.dylib:openssl
+
+startupitem.create      yes
+startupitem.autostart   yes
+startupitem.executable  ${prefix}/sbin/smtpd -F
+
+add_users           _smtpd group=_smtpd realname=OpenSMTP\ Daemon \
+                    _smtpq group=_smtpq realname=OpenSMTP\ Queue
+
+set sysconfdir ${prefix}/etc/${name}
+
+post-patch {
+    reinplace "s|file:/etc/mail/aliases|file:${sysconfdir}/aliases|g" ${worksrcpath}/usr.sbin/smtpd/smtpd.conf
+}
+
+use_autoreconf      yes
+autoreconf.cmd      ./bootstrap
+
+
+depends_build-append \
+                    port:autoconf \
+                    port:automake \
+                    port:libtool
+
+configure.args      --sysconfdir=${sysconfdir} \
+                    --with-auth-pam \
+                    --with-libevent=${prefix} \
+                    --with-mantype=doc \
+                    --with-path-queue=${prefix}/var/spool/smtpd \
+                    --with-table-db
+
+# The spool directory *must* be installed with uid/gid 0.
+# OpenSMTPD checks these permissions at startup (queue_backend.c, using ckdir
+# from util.c), and aborts if they are not correctly set. Consequently,
+# opensmtpd must be installed as root
+destroot.asroot     yes
+destroot.keepdirs   ${destroot}${prefix}/var/spool/smtpd
+
+post-destroot {
+    xinstall -o root -g wheel -m 711 -d ${destroot}${prefix}/var/spool/smtpd
+    file rename ${destroot}${sysconfdir}/smtpd.conf ${destroot}${sysconfdir}/smtpd.conf.dist
+
+    file link -symbolic ${destroot}${prefix}/sbin/sendmail smtpctl
+    file link -symbolic ${destroot}${prefix}/sbin/mailq smtpctl
+    file link -symbolic ${destroot}${prefix}/sbin/makemap smtpctl
+    file link -symbolic ${destroot}${prefix}/sbin/newaliases smtpctl
+}
+
+post-activate {
+    if {![file exists ${sysconfdir}/smtpd.conf]} {
+        file copy ${sysconfdir}/smtpd.conf.dist ${sysconfdir}/smtpd.conf
+    }
+    if {![file exists ${sysconfdir}/aliases] && [file exists /etc/postfix/aliases]} {
+        file copy /etc/postfix/aliases ${sysconfdir}/aliases
+    }
+}
+
+notes "
+    Both OpenSMTPD and the built-in Postfix daemon listen on port 25 by\
+    default. The easiest way to avoid this conflict is to change the\
+    inet_interfaces setting in /etc/postfix/main.cf to keep Postfix from\
+    listening to the same interfaces as OpenSMTPD, and then restarting Postfix.\
+    To keep Postfix from listening on any interfaces, it is sufficient to set:
+
+        inet_interfaces =
+
+    Please see postconf(5) for more details.
+"
+
+livecheck.url       https://opensmtpd.org/archives/
+livecheck.type      regex
+livecheck.regex     opensmtpd-(\[0-9p.\]+)\\.tar\\.gz
```

Comment:
homepage, livecheck and master_sites are usually set by `github.setup`; since you're setting all three of them manually, `github.setup` seems to only set `name` and `version`. Maybe it would be simpler to just drop the use of the GitHub portgroup if you're not using it for anything? Alternatively, maybe the default livecheck added by the GitHub portgroup already works and you wouldn't need these?